src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -119,13 +119,74 @@ private static bool CanSupportObjectInitializer(ISymbol symbol)
 
         if (symbol is IFieldSymbol fieldSymbol)
         {
-            return !fieldSymbol.Type.IsStructType();
+            return MemberTypeCanSupportObjectInitializer(fieldSymbol.Type);
         }
         else if (symbol is IPropertySymbol propertySymbol)
         {
-            return !propertySymbol.Type.IsStructType();
+            return MemberTypeCanSupportObjectInitializer(propertySymbol.Type);
         }
 
         throw ExceptionUtilities.Unreachable();
     }
+
+    private static bool MemberTypeCanSupportObjectInitializer(ITypeSymbol type)
+    {
+        // NOTE: While in C# it is legal to write 'Member = {}' on a member of any of
+        // the ruled out types below, it has no effects and is thus a needless recommendation
+        // Example of the above case:
+        /*
+            class C
+            {
+                string S { get; }
+            }
+
+            new C()
+            {
+                S = {},
+            };
+        */
+
+        // We avoid some types that are common and easy to rule out
+        switch (type.SpecialType)
+        {
+            case SpecialType.System_Enum:
+            case SpecialType.System_String:
+            case SpecialType.System_Object:
+            case SpecialType.System_Delegate:
+            case SpecialType.System_MulticastDelegate:
+
+            // We cannot use collection initializers in Array members,
+            // but for members of an array type with a specified rank we can
+            // For example, assuming Array2D is int[,]:
+            // Array2D = { [0, 0] = value, [0, 1] = value1 },
```

Comment:
I don't understand this comment at all. It seems like the comment is saying you can do this, but then we're still filtering out the member by returning false?

Reply:
If Array2D was `Array`, like so: `Array Array2D = new int[2, 2];`, then we would not be able to use a collection initializer. I should say "symbols of type Array" to further clarify that.

Reply:
Applied changes to the comment